src/extensions/mountainview/MVSortingView/VisibleUnitsControl.tsx

```diff
@@ -0,0 +1,67 @@
+import { Checkbox } from '@material-ui/core';
+import React, { FunctionComponent, useCallback, useEffect, useState } from 'react';
+import { useHitherJob } from '../../common/hither';
+import { Recording, Sorting, SortingSelection, SortingSelectionDispatch } from '../../extensionInterface';
+
+type Props = {
+    sorting: Sorting
+    recording: Recording
+    selection: SortingSelection
+    selectionDispatch: SortingSelectionDispatch
+}
+
+type PeakElectrodeIds = {[key: string]: number}
+
+const VisibleUnitsControl: FunctionComponent<Props> = ({ sorting, recording, selection, selectionDispatch }) => {
+    const sortingInfo = sorting.sortingInfo
+    const [hideRejected, setHideRejected] = useState(false)
+    const [showAcceptedOnly, setShowAcceptedOnly] = useState(false)
+    const [restrictToVisibleElectrodes, setRestrictToVisibleElectrodes] = useState(true)
+    const visibleElectrodeIds = selection.visibleElectrodeIds
+
+    const {result: peakElectrodeIds} = useHitherJob<PeakElectrodeIds>('createjob_get_peak_channels', {sorting_object: sorting.sortingObject, recording_object: recording.recordingObject}, {useClientCache: true})
+
+    const handleShowAcceptedOnly = useCallback(() => {
+        setShowAcceptedOnly(!showAcceptedOnly)
+    }, [showAcceptedOnly])
+    const handleHideRejected = useCallback(() => {
+        setHideRejected(!hideRejected)
+    }, [hideRejected])
+    const handleRestrictToVisibleElectrodes = useCallback(() => {
+        setRestrictToVisibleElectrodes(!restrictToVisibleElectrodes)
+    }, [restrictToVisibleElectrodes])
+    useEffect(() => {
+        const includeUnit = (uid: number) => {
+            if ((restrictToVisibleElectrodes) && (visibleElectrodeIds) && (peakElectrodeIds)) {
+                const peakElectrodeId = peakElectrodeIds[uid + '']
+                if (!visibleElectrodeIds.includes(peakElectrodeId)) return false
+            }
+            if (showAcceptedOnly) {
+                return getLabelsForUnitId(uid, sorting).includes('accept')
+            }
+            else if (hideRejected) {
+                return !getLabelsForUnitId(uid, sorting).includes('reject')
```

Comment:
What's the current status of supporting user-defined annotations? (Wondering about the possibility of typos for these labels.)

Reply:
it's a good question - I'm not sure the best way here... we'll think about it